control/control_loop.go

```diff
@@ -113,31 +114,40 @@ func createLoop(logger logging.Logger, cfg Config, m Controllable) (*Loop, error
 			l.activeBackgroundWorkers.Add(1)
 			utils.ManagedGo(func() {
 				b := b
-				nInputs := len(b.ins)
 				close(waitCh)
 				for {
-					sw := make([]*Signal, nInputs)
-					for i, c := range b.ins {
+					sw := []*Signal{}
+					s := []*Signal{}
+					for _, c := range b.ins {
 						r, ok := <-c
 						if !ok {
 							b.mu.Lock()
 							for _, out := range b.outs {
 								close(out)
 							}
-							// logger.Debugf("Closing outs for block %s %+v\r\n", b.blk.Config(ctx).Name, r)
 							b.outs = nil
 							b.mu.Unlock()
 							return
 						}
-						if len(r) == 1 {
-							sw[i] = r[0]
```

Comment:
Link this diff in a url so we can use it as a base for making a multidimensional signal. I think there's a combo of nInputas and nOuts that can make this workable, as well as adding some typing to the control block configs. But I'll look at that late this week.

Reply:
will do